### DISCUSSION TITLE
Cefsharp compatible with non-default AppDomains

### DISCUSSION THREAD

**Discussion by andre257** [cefsharp/CefSharp#4113] [Q&A] [answered]
Hello guys!

Please, what do we need to do to include this feature in the main branch?

https://github.com/cefsharp/CefSharp/pull/1556

I have a very large wpf app that has a lot of implementations in the region where the browser is launched and this issue of working with two app domains in the same program is making it difficult for my app to improve.

If I could use Cefsharp without implementing the app-domain with add-in, that would be very good and would make the development of new features much easier.

Can you please help me to continue this issue?

**Comment by amaitland** [comment, marked as answer]
> Please, what do we need to do to include this feature in the main branch?

App domains are no longer supported in newer versions of .Net as per https://docs.microsoft.com/en-us/dotnet/core/porting/net-framework-tech-unavailable#application-domains

Currently there's no plan to support CefSharp in non default app domains for older .Net versions. 

As for building your own custom branch, the PR is too old, the project changed too much for me to provide any guidance there.

> Can you please help me to continue this issue?

Your options are

- Run CefSharp out of process using say https://github.com/red-gate/RedGate.AppHost there's an old example at https://github.com/stever/AppHostCefSharp
- Try a different .Net browser.  There are multiple alternatives available.

**Reply by amaitland** [reply]
> Run CefSharp out of process using say https://github.com/red-gate/RedGate.AppHost there's an old example at https://github.com/stever/AppHostCefSharp

There's a work in process for a `CefSharp` specific out of process implementation now available at https://github.com/cefsharp/CefSharp.OutOfProcess
